src/services/recipe.service.ts

```diff
@@ -73,10 +71,20 @@ export class RecipeService {
     return null;
   }
 
+  public async deleteRecipeImages(recipeId: string): Promise<boolean> {
+    const count = await Image.countDocuments({ recipeId });
+    const deleted = await Image.deleteMany({ recipeId });
+    const deletedCount = deleted?.deletedCount;
+    if (count === deletedCount) {
+      return true;
+    }
+    return false;
+  }
+
   public async deleteRecipeById({
     recipeId,
     userId,
-  }: DeleteRecipe): Promise<boolean> {
+  }: DeleteRecipe): Promise<boolean | string> {
```

Comment:
I'd convert the return type to some object instead of `boolean | string`.

For example
```ts
interface DeleteRecipeResult {
  recipeDeleted: boolean;
  recipeImagesDeleted: boolean;
}
```

It's better than checking the result type of the response 
```ts
const res = await this.deleteRecipeById({recipeId, userId});
if (typeof res === 'string') {
  // do something with string
} else {
 // do something with boolean
}
```